### PR TITLE
fix: 178 stz-feat, stz-boost not used within decorator

### LINKED ISSUE #178: [Node SDK] stz-feat, stz-boost not used within guard
stz-feat and stz-boost baggage headers are properly propagated through the system but they are not being used when requesting for quota or sending metrics

### PR DESCRIPTION
**Pull request by ktrz-sre-is**
fixes #178

### CODE REVIEW

File: packages/node/src/guard/stanzaGuard.test.ts [hunk 1]
```diff
@@ -448,6 +450,300 @@ describe('stanzaGuard', function () {
...
+        const contextWithBaggage = new StanzaBaggagePropagator().extract(ROOT_CONTEXT, {
+          baggage: 'stz-boost=1'
+        }, {
+          keys (carrier) {
+            return Object.keys(carrier)
+          },
+          get (carrier, key) {
+            return carrier[key]
+          }
+        })
```
Comment (jamie-kuppens-sre-is) [inline packages/node/src/guard/stanzaGuard.test.ts:556]: Suggestion: I've seen these functions defined quite a few times. Perhaps we can make a helper function for this?

File: packages/node/src/context/priorityBoost/getPriorityBoostFromContext.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { type Context } from '@opentelemetry/api'
+import { stanzaPriorityBoostContextKey } from './stanzaPriorityBoostContextKey'
+export const getPriorityBoostFromContext = (context: Context): number => {
+  const contextPriorityBoostValue = context.getValue(stanzaPriorityBoostContextKey)
+  return typeof contextPriorityBoostValue === 'number' && !isNaN(contextPriorityBoostValue) ? contextPriorityBoostValue : 0
+}
```
Comment (jamie-kuppens-sre-is) [inline packages/node/src/context/priorityBoost/getPriorityBoostFromContext.ts:6]: Same suggestion as my previous comment about `parseInt` and `Number`

**Review by jamie-kuppens-sre-is — APPROVED** [review]

File: packages/node/src/baggage/priorityBoost/getPriorityBoostFromContextBaggage.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { type Baggage, type Context, propagation } from '@opentelemetry/api'
+import { getStanzaBaggageEntry } from '../getStanzaBaggageEntry'
+
+const getStanzaPriorityBoost = (baggage: Baggage) => {
+  const baggageMaybePriorityBoost = parseInt(
+    getStanzaBaggageEntry('stz-boost', baggage)?.value ?? ''
+  )
+
+  return !isNaN(baggageMaybePriorityBoost) ? baggageMaybePriorityBoost : 0
+}
```
[the remark this answers is not in the document]
Comment (ktrz-sre-is) [reply packages/node/src/baggage/priorityBoost/getPriorityBoostFromContextBaggage.ts:10]: We actually expect the returned value to be an integer so I think the slight inconvenience of checking for NaN makes sense here

File: packages/node/src/context/priorityBoost/getPriorityBoostFromContext.ts (hunk 2 above)
Comment (ktrz-sre-is) [reply packages/node/src/context/priorityBoost/getPriorityBoostFromContext.ts:6]: in this case `context.getValue` returns `unknown` type so we just need to verify that it's actually a number.

File: packages/node/src/utils/wrapEventsAsync.ts [hunk 4]
```diff
@@ -1,19 +1,40 @@
...
+  duration?: (
+    duration: number,
+    result: TResult | undefined,
+    ...args: TArgs
+  ) => void | Promise<void>
+}
+
+type AsyncLikeFn<TArgs extends any[], TResult> = (
+  ...args: TArgs
+) => PromiseLike<TResult>
```
[the remark this answers is not in the document]
Comment (ktrz-sre-is) [reply packages/node/src/utils/wrapEventsAsync.ts:13]: this was an unnecessary complication. removed

File: packages/node/src/guard/stanzaGuard.test.ts (hunk 1 above)
Comment (ktrz-sre-is) [reply packages/node/src/guard/stanzaGuard.test.ts:556]: done

**Pull request merged**